### PR TITLE
[Azure.Core] add test helper for Models

### LINKED ISSUE #35740: [Azure.Core] Add helper method for comparing two Models
Add generic model checker that verifies all model properties - can verify that two models are the same

### PR DESCRIPTION
**Pull request by nisha-bhatia**
Fixes: https://github.com/Azure/azure-sdk-for-net/issues/35740

# Contributing to the Azure SDK

Please see our [CONTRIBUTING.md](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md) if you are not familiar with contributing to this repository or have questions.

For specific information about pull request etiquette and best practices, see [this section](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#pull-request-etiquette-and-best-practices).

### CODE REVIEW

**Comment by nisha-bhatia** [comment]
https://github.com/Azure/azure-sdk-for-net/issues/35740

File: sdk/core/Azure.Core/tests/ModelSerializationTests/VerifyModels.cs [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+
+        public static void CheckDogs(DogListProperty x, DogListProperty y, SerializableOptions options)
+        {
+            if (options.IgnoreReadOnlyProperties)
+                Assert.That(x.LatinName, Is.EqualTo(y.LatinName));
+            Assert.That(x.Name, Is.EqualTo(y.Name));
+            Assert.That(x.Weight, Is.EqualTo(y.Weight));
+            Assert.That(x.FoodConsumed, Is.EqualTo(y.FoodConsumed));
+
+            if (options.IgnoreReadOnlyProperties)
```
Comment (m-nash) [inline sdk/core/Azure.Core/tests/ModelSerializationTests/VerifyModels.cs:75]: should this be if not ignoreadditonalproperties?

File: sdk/core/Azure.Core/tests/ModelSerializationTests/VerifyModels.cs [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+                {
+                    Assert.IsTrue(additionalPropertiesX.ContainsKey(additionalProperty.Key));
+                    Assert.AreEqual(additionalProperty.Value.ToString(), additionalPropertiesX[additionalProperty.Key].ToString());
+                }
+            }
+        }
+
+        public static void CheckDogs(DogListProperty x, DogListProperty y, SerializableOptions options)
+        {
+            if (options.IgnoreReadOnlyProperties)
```
Comment (m-nash) [inline sdk/core/Azure.Core/tests/ModelSerializationTests/VerifyModels.cs:69]: shouldn't this be if not ignorereadonly properties?

**Review by christothes — COMMENTED** [review]
could we generalize this with a generic `ModelTypesAreEquivalent<T>`

**Comment by nisha-bhatia** [comment]
> could we generalize this with a generic `ModelTypesAreEquivalent<T>`

I'm not sure about generalizing it, since each model has specific properties I need to verify but I will create a helper so there is less duplicated code.

**Review by m-nash — APPROVED** [review]

**Pull request merged**